ni_measurementlink_service/_internal/parameter/serializer.py

```diff
@@ -158,7 +158,9 @@ def _get_overlapping_parameters(
             )
         type = parameter_metadata_dict[field_index].type
         is_repeated = parameter_metadata_dict[field_index].repeated
-        decoder = serialization_strategy.Context.get_decoder(type, is_repeated)
+        decoder = serialization_strategy.Context.get_decoder(
+            type, is_repeated, parameter_metadata_dict[field_index].message_type
+        )
         inner_decoder = decoder(field_index, field_index)
```

Comment:
The 2nd parameter is `key`, which is typed as `str` in `serialization_strategy.py`. 

Here, we are passing an `int`.

typeshed annotates `key` as a `FieldDescriptor`. 

Are these the same thing? Which is right?